tests_integration/__tests__/plugin-resolution.js

```diff
@@ -173,7 +173,20 @@ describe("loads --plugin by relative path to its directory without leading ./ (a
     "--parser=bar",
     "--plugin=automatic/node_modules/prettier-plugin-bar"
   ]).test({
-    stdout: "bar+contents" + EOL,
+    stdout: "content from `prettier-plugin-bar` package + contents" + EOL,
+    stderr: "",
+    status: 0,
+    write: []
+  });
+});
+
+describe("loads --plugin by filename without leading ./ and ext, should resolve to file, not package", () => {
+  runPrettier("plugins/automatic", [
+    "file.txt",
+    "--parser=bar",
+    "--plugin=prettier-plugin-bar"
```

Comment:
This part do the trick, if we don't try local file first, will resolve to the package `prettier-plugin-bar` , instead of `prettier-plugin-bar.js` file

Reply:
I would favor documenting this in the test itself.